### PR TITLE
Use init_scope in SequentialFeatureExtractor

### PR DESCRIPTION
**Pull request by Hakuyume**
`OrderedDict` is similar to `**links` of `chainer.Chain.__init__`. This style is not recommended since Chainer v2.

### CODE REVIEW

**Comment by yuyu2172** [comment]
The change seems to make inheriting this class nice. The inheriting class can register links in the style recommended in v2.

Not all classes need to have `chainer.Chain` like inheritance (e.g. `chainercv.links.FasterRCNN` and `chainercv.links.SSD`), but it seems that a general one like this can be like that.

File: chainercv/links/model/sequential_feature_extractor.py [hunk 1]
```diff
@@ -33,42 +31,47 @@ class SequentialFeatureExtractor(chainer.Chain):
...
-            raise ValueError('layers need to be collections.OrderedDict')
-        self._layers = layers
-
-        self.layer_names = layer_names
-
-        with self.init_scope():
-            for name, layer in self._layers.items():
-                if isinstance(layer, chainer.Link):
-                    setattr(self, name, layer)
+        self._order = list()
```
Comment (yuyu2172) [inline chainercv/links/model/sequential_feature_extractor.py:57]: Why do you prefer `_layers` over `_order`?
`_order` sounds like a list of integers. It doe not imply that it contains links and functions.

Comment (yuyu2172) [reply chainercv/links/model/sequential_feature_extractor.py:57]: Sorry. I realized that it is a list of names.

Comment (Hakuyume) [reply chainercv/links/model/sequential_feature_extractor.py:57]: `_layers` sounds that it contains callable objects. I think `_layer_names` is the best name, but it conflicts with an existing member. As you pointed out, `_order` is not good.

**Comment by yuyu2172** [comment]
Some tests failed on GPU mode.

File: chainercv/links/model/sequential_feature_extractor.py (hunk 1 above)
Comment (yuyu2172) [reply chainercv/links/model/sequential_feature_extractor.py:57]: I think `_order` is not that bad.

File: chainercv/links/model/sequential_feature_extractor.py [hunk 2]
```diff
@@ -9,11 +7,11 @@ class SequentialFeatureExtractor(chainer.Chain):
 
     This class is a base class that can be used for an implementation of
     a feature extractor model.
-    The link takes an argument :obj:`layers` that specifies the computation
-    conducted in :meth:`__call__`.
-    :obj:`layers` is a :class:`collections.OrderedDict` of
-    callable objects called layers, which are going to be called sequentially
-    starting from the top to the bottom.
+    Callabel objects, such as :class:`chainer.Link` and
```
Comment (yuyu2172) [inline chainercv/links/model/sequential_feature_extractor.py]: s/Callabel/Callable

File: chainercv/links/model/sequential_feature_extractor.py [hunk 3]
```diff
@@ -9,11 +7,11 @@ class SequentialFeatureExtractor(chainer.Chain):
...
+    :class:`chainer.Function`, can be registered to this link with
```
Comment (yuyu2172) [inline chainercv/links/model/sequential_feature_extractor.py]: This is a problem from the original doc, but can you change the phrase `this link` to `this chain`?

File: chainercv/links/model/sequential_feature_extractor.py [hunk 4]
```diff
@@ -9,11 +7,11 @@ class SequentialFeatureExtractor(chainer.Chain):
...
+    :meth:`init_scope`.
+    This link keeps the order of registerations and conducts the computation
```
Comment (yuyu2172) [inline chainercv/links/model/sequential_feature_extractor.py]: :meth:`__call__` executes callables in the order in which they are registered to.

**Comment by Hakuyume** [comment]
> Some tests failed on GPU mode.

Sorry, I fixed the tests.

File: chainercv/links/model/sequential_feature_extractor.py [hunk 5]
```diff
@@ -9,13 +7,13 @@ class SequentialFeatureExtractor(chainer.Chain):
...
-    The link takes an argument :obj:`layers` that specifies the computation
-    conducted in :meth:`__call__`.
-    :obj:`layers` is a :class:`collections.OrderedDict` of
-    callable objects called layers, which are going to be called sequentially
-    starting from the top to the bottom.
+    Callable objects, such as :class:`chainer.Link` and
+    :class:`chainer.Function`, can be registered to this chain with
+    :meth:`init_scope`.
+    This link keeps the order of registerations and :meth:__call__
+    executes callables in the order in which they are registered to.
```
Comment (yuyu2172) [inline chainercv/links/model/sequential_feature_extractor.py]: Sorry. The sentence I proposed was wrong. BTW, You forgot ```.

Either,
+ :meth`__call__` executes callables in the order in which they are registered to this chain.
+ This chain keeps the order of registrations and :meth:`__call__` executes callables in that order.

**Comment by yuyu2172** [comment]
LGTM

File: chainercv/links/model/sequential_feature_extractor.py (hunk 1 above)
Comment (Hakuyume) [reply chainercv/links/model/sequential_feature_extractor.py:57]: OK, I use `_order`.

**Pull request merged**